docs/core/build/distribution-packaging.md

```diff
@@ -70,17 +93,62 @@ For example: SDK version 2.2.302 is the second patch release of the third featur
 
-The following table shows the recommended packages:
-
-| Name                                    | Example                | Use case: Install ...           | Contains           | Dependencies                                   | Version            |
-|-----------------------------------------|------------------------|---------------------------------|--------------------|------------------------------------------------|--------------------|
-| dotnet-sdk-[major]                      | dotnet-sdk-2           | Latest sdk for runtime major    |                    | dotnet-sdk-[major].[latestminor]               | \<sdk version>     |
-| dotnet-sdk-[major].[minor]              | dotnet-sdk-2.1         | Latest sdk for specific runtime |                    | dotnet-sdk-[major].[minor].[latest sdk feat]xx | \<sdk version>     |
-| dotnet-sdk-[major].[minor].[sdk feat]xx | dotnet-sdk-2.1.3xx     | Specific sdk feature release    | (3),(4)            | aspnetcore-runtime-[major].[minor]             | \<sdk version>     |
-| aspnetcore-runtime-[major].[minor]      | aspnetcore-runtime-2.1 | Specific ASP.NET Core runtime   | (6),[(7)]          | dotnet-runtime-[major].[minor]                 | \<runtime version> |
-| dotnet-runtime-[major].[minor]          | dotnet-runtime-2.1     | Specific runtime                | (5)                | host-fxr:\<runtime version>+                   | \<runtime version> |
-| dotnet-host-fxr                         | dotnet-host-fxr        | _dependency_                    | (2)                | host:\<runtime version>+                       | \<runtime version> |
-| dotnet-host                             | dotnet-host            | _dependency_                    | (1),(8),(9),(10)   |                                                | \<runtime version> |
+The following lists the recommended packages:
+
+* `dotnet-sdk-[major]` - Installs the latest sdk for runtime major
+  * **Version:** \<sdk version>
+  * **Example:** dotnet-sdk-2
+  * **Dependencies:** `dotnet-sdk-[major].[latestminor]`
+
+* `dotnet-sdk-[major].[minor]` - Installs the latest sdk for specific runtime
+  * **Version:** \<runtime version>
+  * **Example:** dotnet-sdk-2.1
+  * **Contains:** (3),(4)
+  * **Dependencies:** `aspnetcore-runtime-[major].[minor]`, `dotnet-netcoreapp-targeting-pack[major].[minor]`, `dotnet-aspnetcore-targeting-pack[major].[minor]`, `dotnet-netstandard-targeting-pack-[netstandard_major].[netstandard_minor]`, `dotnet-apphost-pack-[major].[minor]`
+
+* `aspnetcore-runtime-[major].[minor]` - Installs specific ASP.NET Core runtime
+  * **Version:** \<aspnetcore runtime version>
+  * **Example:** aspnetcore-runtime-2.1
+  * **Contains:** (6),(7)
+  * **Dependencies:** `dotnet-runtime-[major].[minor]`
+
+* `dotnet-runtime-deps-[major].[minor]` - Installs the dependencies for running self-contained applications
```

Comment:
I have  mixed feelings about this. This package was initially introduced so all other dotnet packages would depend on this one. And only `dotnet-runtime-deps*` would have OS-specific dependencies.

Is the description here that the goal is to "Installs the dependencies for running self-contained applications" accurate?

Reply:
It's not the description of the package. In the table, this was the "Use case" why someone might want to install this.

Reply:
This is interesting... for the Microsoft build, it serves both purposes, but I don't know whether the "use one package for multiple distros/versions" approach is used for typical Linux distro packaging. *If* someone wants to build the SDK once to patch multiple versions of a distro, they'd hit the same problem as the Microsoft build and might want a hint in this doc that only the deps package has to be unique per version.

Reply:
With my package maintainer hat on, this package might be trickier to provide than expected. RPM, for example, automatically identifies (using ldd, I guess) all the dependencies. As a packager, I supplement it with stuff that RPM can't see (for example, libraries loaded via `dlopen(3)`). [This is what that looks like for .NET Core Runtime 2.1](https://gist.github.com/omajid/25ab3aaa4b3bf178d1a8fe5c112aba59). Having a `runtime-deps` package means moving all those dependencies to a new package. RPM the package builder has no support for that, really. I would have to disable the automated dependency detection (which is against the Fedora packaging guidelines) and then manually identify all the dependencies and write them down in a different place. And then keep that up-to-date as .NET Core evolves.

This creates more work for a packager, makes the process more error prone, and goes against Linux distribution guidelines.

Can we have `dotnet-runtime-deps-[major].[minor]` be optional? As in, someone can provide it if they really want to, but it's not required (or strongly suggested)?

Reply:
It seems to me that a self-contained app developer/user would like to be able to install `dotnet-runtime-deps-x-y` on any "has .NET Core" distro with some amount of certainty... rather than install bloat (a full `dotnet-runtime-x-y`), or maintain a list of dependencies for each distro they build for. This seems like it could be important for space-constrained scenarios.

Definitely want feedback from @leecow on this point I think. Very interesting to hear how this makes things more difficult.

Reply:
We can mention it is optional.
I think the main value of the package is that it makes it easy to make a container image that is small by including `dotnet-runtime-deps` and a self-contained application.
It's not a use-case that is super important to us (Red Hat) afaik.

Reply:
Hmm, yeah, it seems reasonable to me to mark it optional with a description of when a package maintainer would want to provide it.

Reply:
Sorry for being late to the party. I agree with the idea of describing the deps package as optional along with the use case.